inspirehep/dojson/hep/rules/bd6xx.py

```diff
@@ -62,133 +61,86 @@ def accelerator_experiments2marc(self, key, value):
-
-    def _freekey_get_source(source_value):
-        if not isinstance(source_value, (list, tuple, set)):
-            return source_value
-
-        source_value = [source.lower() for source in source_value]
-        if 'conference' in source_value:
-            return ''
-
-        for source in source_value:
-            if source in ['author', 'publisher']:
-                return source
-
-        return ''
-
-    def _freekey_get_dict(elem):
-        keyword = {
-            'value': elem.get('a'),
-        }
-
-        source = _freekey_get_source(elem.get('9'))
-        if source:
-            keyword['source'] = source
-
-        return keyword
-
-    def _get_keyword_dict(key, elem):
-        if _is_thesaurus(key) and elem.get('a'):
-            _schema = load_schema('hep')
-            valid_keywords = _schema['properties']['keywords']['items']['properties']['schema']['enum']
-            schema_in_marc = elem.get('2').upper()
-            schema = schema_in_marc if schema_in_marc in valid_keywords else ''
-
-            return {
-                'schema': schema,
-                'source': elem.get('9', ''),
-                'value': elem.get('a'),
-            }
-        elif _is_energy(elem):
-            return {}
-        else:
-            return _freekey_get_dict(elem)
-
-    if _is_energy(value):
-        if 'energy_ranges' not in self:
-            self['energy_ranges'] = list()
-
-        self['energy_ranges'].append(int(value.get('e')))
-        self['energy_ranges'].sort()
-
-    result = _get_keyword_dict(key, value)
-    if 'value' in result and result['value']:
-        return result
-
-
-@hep2marc.over('695', '^keywords$')
-def keywords2marc(self, key, values):
-    values = force_list(values)
-
-    def _is_thesaurus(elem):
-        return elem.get('schema', '') is not ''
-
-    def _thesaurus_to_marc_dict(elem):
-        result = {
-            'a': elem.get('value'),
-            '2': elem.get('schema'),
-        }
-        source = elem.get('source', None)
-        if source:
-            result['9'] = elem.get('source')
-
-        return result
-
-    def _freekey_to_marc_dict(elem):
-        return {
-            'a': elem.get('value'),
-            '9': elem.get('source', ''),
-        }
-
-    thesaurus_terms = self.get('695', [])
-
-    for value in values:
-        if _is_thesaurus(value):
-            marc_dict = _thesaurus_to_marc_dict(value)
-            thesaurus_terms.append(marc_dict)
-        else:
-            marc_dict = _freekey_to_marc_dict(value)
-            if '653' not in self:
-                self['653'] = []
-
-            if marc_dict.get('9') != 'magpie':
-                self['653'].append(marc_dict)
-            continue
+    keywords = self.get('keywords', [])
+    energy_ranges = self.get('energy_ranges', [])
+
+    for value in force_list(values):
+        if value.get('e'):
+            energy_ranges.append(maybe_int(value.get('e')))
+        elif value.get('a'):
```

Comment:
are we sure there is never `a` and `e` in the same `695` field? maybe `if` instead of `elif` to play it safe?